spectator-reg-micrometer/src/main/java/com/netflix/spectator/micrometer/MicrometerRegistry.java

```diff
@@ -74,12 +74,16 @@ private Id convert(io.micrometer.core.instrument.Meter.Id id) {
 
   private Meter convert(io.micrometer.core.instrument.Meter meter) {
     Id id = convert(meter.getId());
-    switch (meter.getId().getType()) {
-      case COUNTER:              return counter(id);
-      case TIMER:                return timer(id);
-      case DISTRIBUTION_SUMMARY: return distributionSummary(id);
-      case GAUGE:                return gauge(id);
-      default:                   return null;
+    if (meter instanceof io.micrometer.core.instrument.Counter) {
+      return counter(id);
+    } else if (meter instanceof io.micrometer.core.instrument.Timer) {
+      return timer(id);
+    } else if (meter instanceof io.micrometer.core.instrument.DistributionSummary) {
+      return distributionSummary(id);
+    } else if (meter instanceof io.micrometer.core.instrument.Gauge) {
+      return gauge(id);
+    } else {
+      return null;
```

Comment:
I decided to remove the switch based on Micrometer's `Type` since that doesn't correspond always to the Meter type one would assume.